### PR TITLE
chore: Allows to to run acceptance tests for a published Atlas provider version

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Allows to to run acceptance tests for a published Atlas provider version.

This allows to run acceptance test for already published MongoDB Atlas provider releases for a Terraform CLI version and check that it's compatible.

Link to any related issue(s): https://jira.mongodb.org/browse/CLOUDP-219089

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [X] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [X] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code
- [X] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: .github/workflows/test-suite.yml [hunk 1]
```diff
@@ -34,8 +34,16 @@ jobs:
     secrets: inherit
     uses: ./.github/workflows/cleanup-test-env.yml
 
+  acc-tests-latest-published:
+    needs: clean-before
+    if: ${{ github.event_name == 'schedule' && !cancelled() }}
+    secrets: inherit
+    uses: ./.github/workflows/acceptance-tests.yml
+    with:
+      ref: 'v1.14.0'
```
Comment (lantoli) [inline .github/workflows/test-suite.yml]: pending to calculate instead of hardcoding

Comment (oarbusi) [inline .github/workflows/test-suite.yml:43]: Will we need to change this value with every release?

File: .github/workflows/test-suite.yml [hunk 2]
```diff
@@ -34,8 +34,16 @@ jobs:
     secrets: inherit
     uses: ./.github/workflows/cleanup-test-env.yml
 
+  acc-tests-latest-published:
```
Comment (AgustinBettati) [inline .github/workflows/test-suite.yml]: I am struggling to see the value of adding this job to run daily, given that the results will be the same unless there is a new patch release from TF. 
If you can provide more context in the PR or reference where this requirement was discussed that would be great. I am thinking why we cannot rely on our master branch to see if the latest patch version of TF is breaking some functionality in our provider.

Comment (lantoli) [reply .github/workflows/test-suite.yml]: the use case here is that Hashicorp releases a new version, e.g. 1.6.7 and breaks our latest published version, e.g. 1.14.0.    it's true that results won't change while Hashicorp doesn't publish a new version. But the alternative is that we have to be pending (or automate?) when a new version is published and then run on demand.

Comment (lantoli) [reply .github/workflows/test-suite.yml]: @Zuhairahmed what do you think?
alternatively we can do that, don't run every night but on demand when we know that a new Hashicorp version came out.

Comment (lantoli) [reply .github/workflows/test-suite.yml]: @AgustinBettati at the moment I'll leave only the option to run it manually

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Pull request merged**